python/lsst/meas/extensions/multiprofit/consolidate_astropy_table.py

```diff
@@ -196,18 +210,32 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
 
         tables_read = []
         check_columns = self.config.drop_duplicate_columns or self.config.validate_duplicate_columns
+        n_bands = len(bands_sorted)
 
         for name, data_name in data.items():
             config_input = self.config.inputs[name]
-            tables = [
-                (
-                    apTab.hstack([data_name[patch][band] for band in bands_sorted], join_type="exact")
-                    if not config_input.is_multiband
-                    else data_name[patch][band_null]
-                )
-                for patch in (patches_ref if not config_input.is_multipatch else patches_null)
-            ]
-            table_new = tables[0] if (len(tables) == 1) else apTab.vstack(tables, join_type="exact")
+            tables = []
+            bands_missing = False
+            for patch in (patches_ref if not config_input.is_multipatch else patches_null):
+                data_name_patch = data_name[patch]
+                if config_input.is_multiband:
+                    tables_patch = data_name_patch.get(band_null, [])
+                else:
+                    tables_patch = [
+                        _tab for band in bands_sorted if (_tab := data_name_patch.get(band)) is not None
+                    ] if not config_input.is_multiband else data_name[patch][band_null]
```

Comment:
This is really hard to follow. If you're in this else block `config_input.is_multiband` is DEFINITELY False.

Reply:
Right, I meant to delete the `if ...` part.